### ISSUE TITLE
BLTOUCH pin

### ISSUE THREAD

**Issue by kennethgomez01** [Poket-Jony/Marlin-A4MaxPro-2.0.x#50]
How to wire BLTOUCH to anycubic 4 max pro?

**Comment by Erik262** [comment]
You can use this...

![IMG_2638-min](https://user-images.githubusercontent.com/11977647/135251814-360a16fd-6a55-4d7d-91f4-641bcb50adc9.png)

**Comment by kennethgomez01** [comment]
@Erik262 Hi, is this your printer setup? If it is, can you show me how you put the BLTouch sensor on the hot end? And please tell me the pin of where to plug the bltouch sensor? Thanks.

**Comment by Erik262** [comment]
I put my BLTouch sensor inside of the head case. The wiring is exactly the same as the "I3Mega". Take a closer look on the picture, my connectors are labeled like this one: 
Cable Color (GRB) where is 'G = +' and 'R = -' etc.

**Comment by kennethgomez01** [comment]
> I put my BLTouch sensor inside of the head case. The wiring is exactly the same as the "I3Mega". Take a closer look on the picture, my connectors are labeled like this one: Cable Color (GRB) where is 'G = +' and 'R = -' etc.

Hi, just wondering as it says everytime I compile the Marlin, the current marlin onfile is old and need to be updated. Could you help me how to solve this?

**Comment by Erik262** [comment]
There should be a precompiled version available here. If I'm correct.

**Comment by kennethgomez01** [comment]
even your precompiled is causing error.

**Comment by Erik262** [comment]
> even your precompiled is causing error.

Try this from my repo: https://github.com/Erik262/Marlin-A4MaxPro-2.0.x/tree/bltouch

**Comment by kennethgomez01** [comment]
also why the probe is not extending when I press autoleveling?

**Comment by Erik262** [comment]
Did you set G29 in your start script? Here is an example of mine. But be careful, my settings might not work without any issues. So keep this as an example.

G21 ; set units to millimeters
G90 ; absolute positioning
G1 F3600 ; set standard speed
M107 ; turn all fans off
G28 X Y ; home X and Y axis
G28 Z ; home Z axis
M140 S{material_bed_temperature} ; set bed temperature
M190 S{material_bed_temperature} ; wait for bed temperature
G29 ;autolevel
M500 ;save data
M420 S1 ;retrieve mesh
M300 S1108 P105 ;play melody
M300 S1661 P210 ; play melody
G1 X-5 Y5.0
G1 Z0
M104 S{material_print_temperature} ; set hotend temperature
M109 S{material_print_temperature} ; wait for hotend temperature
G92 E0.0 ; reset extruder distance position
G1 X-5 Y40.0 ; wipe nozzle on brass brush
G1 X-5 Y5.0 ; wipe nozzle on brass brush
G1 X-5 Y40.0 ; wipe nozzle on brass brush
G1 X-5 Y5.0 ; wipe nozzle on brass brush
G1 X0.0 Y5.0 Z0.25 ; set starting point
G1 Y60.0 E4.5 F1000.0 ; print intro line
G1 Y100.0 E21.5 F1000.0 ; print intro line
G92 E0.0 ; reset extruder distance position

**Comment by kennethgomez01** [comment]
on your repo its causing error `ui_api 2.cpp.o (symbol from plugin):(.text+0x0): first defined here
collect2.exe: error: ld returned 1 exit status
exit status 1
Error compiling for board Arduino Mega or Mega 2560.
`

**Comment by kennethgomez01** [comment]
Hi I am just wondering as I am having toruble with z-offser(value is negative) for few days, just want to know is there z offset wizzard for anycbic 4 max pro?

**Comment by kennethgomez01** [comment]
@Erik262  Hi I am just wondering as I am having toruble with z-offser(value is negative) for few days, just want to know is there z offset wizzard for anycbic 4 max pro?

**Comment by Erik262** [comment]
As far as I know, there is no z-offest wizard in that Marlin version.

**Comment by kennethgomez01** [comment]
@Erik262 just question did you remove the z-max switch for bltouch? Because I followed every totorial about z-offset but not be able to succed. I am getting bed not sticking

**Comment by Erik262** [comment]
No, I did not, but if it is touching before the BLTouch does, you have to move it up a bit. You might not save it correctly in your EEPROM or something else.
Try to set the z-offset manually by using a feeler gauge (no paper!) and the wheelknobs under your heatbed. Then do your BLTouch stuff.

**Comment by kennethgomez01** [comment]
@Erik262 can you take a pciture of your setup of bltouch attach to the extruder?

**Comment by Erik262** [comment]
@kennethgomez01 Here are some pictures of my setup on Thingiverse: https://www.thingiverse.com/thing:4090425

**Comment by kennethgomez01** [comment]
@Erik262 are you using the clone? Why I keep getting print not sticking in the bed in some area of the bed? Can you share me the slicer setting you use, and grid points?

**Comment by Erik262** [comment]
I used to use a clone. Now I'm using the original one. The reason why your prints are not sticking on the bed is probably that your z-offset is not correct or you have some weird concave areas on your printbed. You might have to make a fine grid. I think I set in the settings to 25 points. I'm not sure. Or use a gluestick :)

**Comment by kennethgomez01** [comment]
@Erik262 how do you manage to set z_offset? Mine I change the hot end to v6 but the heatsink is same.

**Comment by Erik262** [comment]
You have to set up your z-offset only once like you wouldn't have a BLTouch. Then correct your z-offset with your BLTouch. If you have the wrong z-off set, you will provide this at any place on your printbed. You can change your offset by using the code M-851 or something else. This should be the easiest part. It's been a few years ago since I set this on my printer. I don't know everything what I did there.

**Comment by kennethgomez01** [comment]
Thanks @Erik262
